woocommerce/woocommerce
# [Experimental] Surface custom fields in My Account pages

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Shows checkout custom field data under:

- My Account > View Orders
- My Account > Addresses
- My Account > Addresses > Edit
- My Account > Account details

Supporting changes:

- Introduced `woocommerce_edit_account_form_fields` above the password field
- Updated `woocommerce_form_field` to post a value when using checkboxes even if unchecked
- Introduced `woocommerce_my_account_after_my_address` and `woocommerce_order_details_after_customer_address` hooks
- Updates to styling

Closes #42779

Screenshots:

![Screenshot 2024-01-19 at 14 59 35](https://github.com/woocommerce/woocommerce/assets/90977/94bf6312-69ae-4a03-920b-f7f19647c7ad)
![Screenshot 2024-01-19 at 14 59 44](https://github.com/woocommerce/woocommerce/assets/90977/4a832968-933d-4446-a62b-33d5479413aa)
![Screenshot 2024-01-19 at 14 59 50](https://github.com/woocommerce/woocommerce/assets/90977/5f71cb47-f69f-44dc-94cc-c8003979e1c5)
![Screenshot 2024-01-19 at 15 00 01](https://github.com/woocommerce/woocommerce/assets/90977/66a5132c-948c-4568-8b5e-7b5e1758409b)



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Before testing, implement the snippet to add custom fields as found in #43449
2. Place an order using a logged in account, making sure to fill out the custom fields during checkout
3. Go to the my account page
4. In the orders section, view the order you just placed and check that custom field values are displayed
5. In the account details section, confirm the custom fields you registered are visible as form fields and editable. Edit the values and confirm they persist. 
6. In the address section, check custom fields are shown.
7. Edit an address. Confirm the form fields for custom fields are shown, and you can edit a value and have it persist.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Moved `woocommerce_edit_account_form` above the password field.
Introduced `woocommerce_my_account_after_my_address` and `woocommerce_order_details_after_customer_address` hooks.

#### Comment 

</details>
